test/upgrade/2.2.0-2.2.1.js

```diff
@@ -0,0 +1,200 @@
+const hre = require("hardhat");
+const { RevertReasons } = require("../../scripts/config/revert-reasons.js");
+const ethers = hre.ethers;
+const { assert, expect } = require("chai");
+const DisputeResolver = require("../../scripts/domain/DisputeResolver");
+const Seller = require("../../scripts/domain/Seller");
+const { calculateContractAddress } = require("../util/utils.js");
+const { mockSeller, mockAuthToken, mockVoucherInitValues } = require("../util/mock");
+
+const {
+  deploySuite,
+  upgradeSuite,
+  upgradeClients,
+  populateProtocolContract,
+  getProtocolContractState,
+  revertState,
+} = require("../util/upgrade");
+const { getGenericContext } = require("./01_generic");
+
+const oldVersion = "v2.2.0";
+const newVersion = "v2.2.1-rc.1";
+//const v2_1_0_scripts = "v2.1.0-scripts";
+
+/**
+ *  Upgrade test case - After upgrade from 2.2.0 to 2.2.1 everything is still operational
+ */
+describe("[@skip-on-coverage] After facet upgrade, everything is still operational", function () {
+  this.timeout(10000000);
+  // Common vars
+  let deployer, rando;
+  let accountHandler;
+  let snapshot;
+  let protocolDiamondAddress, mockContracts;
+
+  // reference protocol state
+  let accountContractState;
+  let preUpgradeEntities;
+
+  before(async function () {
+    try {
+      // Make accounts available
+      [deployer, rando] = await ethers.getSigners();
+
+      let contractsBefore;
+      ({
+        protocolDiamondAddress,
+        protocolContracts: contractsBefore,
+        mockContracts,
+      } = await deploySuite(deployer, oldVersion));
+      //      ({ twinHandler} = contractsBefore);
+
+      // Populate protocol with data
+      preUpgradeEntities = await populateProtocolContract(
+        deployer,
+        protocolDiamondAddress,
+        contractsBefore,
+        mockContracts,
+        oldVersion
+      );
+
+      // Get current protocol state, which serves as the reference
+      // We assume that this state is a true one, relying on our unit and integration tests
+      const protocolContractState = await getProtocolContractState(
+        protocolDiamondAddress,
+        contractsBefore,
+        mockContracts,
+        preUpgradeEntities,
+        oldVersion
+      );
+
+      ({ accountContractState } = protocolContractState);
+
+      // upgrade clients
+      await upgradeClients(newVersion);
+
+      // Upgrade protocol
+      ({ accountHandler } = await upgradeSuite(newVersion, protocolDiamondAddress, {
+        accountHandler: "IBosonAccountHandler",
+        orchestrationHandler: "IBosonOrchestrationHandler",
+      }));
+
+      snapshot = await ethers.provider.send("evm_snapshot", []);
```

Comment:
We have a util function `getSnapshot` that does the same.
Not that makes that big change, but looks a bit nicer and it will be more consistent with other tests.

Reply:
done